### PR TITLE
Add EntitySearchInput Component

### PR DESCRIPTION
**Pull request by kuasha420**
## Summary

Addresses issue #4049 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/EntitySearchInput.js [hunk 1]
```diff
@@ -20,49 +20,125 @@
...
+	const onClose = useCallback( () => {
+		setIsOpen( false );
 	}, [] );
 
-	if ( isActive ) {
+	const onEscapeKeyDown = useCallback(
+		( e ) => {
+			if ( e.keyCode === ESCAPE ) {
+				onClose();
+			}
```
[the remark this answers is not in the document]
Comment (kuasha420) [reply assets/js/components/EntitySearchInput.js]: Hi, initially I implemented the `Enter` key functionality like [this](https://github.com/google/site-kit-wp/pull/4261/commits/221752df0563c7aaec93961774814f0809303f59) but it was kind of a mess, so I have reverted that one and done the current iteration. However, I had to 

* deviate from the IB and put the `onKeyDown` function to the `PostSearcherAutoSuggest` component 
* and pass `onClose` as the prop from the `EntitySearchInput` component.

Let me know which one you prefer. 

Another thing I noticed is, if the searchTerm is not yet debounced and API request is not complete, pressing `Enter` will do nothing, even if the searchTerm is a exact match. Is that something we should handle with lots of duplicate code or leave as is? 

Cheers.

File: assets/js/components/PostSearcherAutoSuggest.js [hunk 2]
```diff
@@ -77,17 +85,41 @@ export default function PostSearcherAutoSuggest( {
...
+		( e ) => {
+			if ( ! unifiedDashboardEnabled ) {
+				return;
+			}
+			if ( e.keyCode === ESCAPE ) {
+				return onClose();
+			}
+
+			if ( e.keyCode === ENTER ) {
+				return onSelectCallback( searchTerm );
```
Comment (eugene-manuilov) [inline assets/js/components/PostSearcherAutoSuggest.js:118]: @kuasha420 we also need to update the `onSelectCallback` callback to make it case-insensative:

```js
const foundMatch = results.find(
	( post ) =>
		post.post_title.toLowerCase() === value.toLowerCase()
);
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @kuasha420. Just one more comment and should be good.

File: assets/js/components/EntitySearchInput.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/components/EntitySearchInput.js]: > Another thing I noticed is, if the searchTerm is not yet debounced and API request is not complete, pressing Enter will do nothing, even if the searchTerm is a exact match. Is that something we should handle with lots of duplicate code or leave as is?

I think this is fine for now.

**Comment by eugene-manuilov** [comment]
@kuasha420 could you please also add the pointer cursor to the search results styles?

**Review by eugene-manuilov — APPROVED** [review]
LGTM! Thanks, @kuasha420.

**Comment by kuasha420** [comment]
@eugene-manuilov fixed the conflicts. Thanks.

**Review by eugene-manuilov — APPROVED** [review]

**Pull request merged**